### PR TITLE
Retry loop for redis-cli BGSAVE

### PR DESCRIPTION
**Pull request by buckelij**
`ghe-backup-redis` dumps can fail and hang indefinitely if BGSAVE is called while Redis "background rewriting" is happening. In these cases BGSAVE will report an error, but still exit 0. 

This PR adds a `for` loop around the `redis-cli BGSAVE` to retry if BGSAVE reports an `ERR`. I also put the `LASTSAVE` check in a `for` loop that times out after 60 minutes. 

I'm not certain how long the `for` loop `sleep`s should be, but I think 60 minutes is more than enough.

### CODE REVIEW

**Comment by buckelij** [comment]
So, `debuild` from `package-deb` was running the tests as well. That seems redundant, and was causing the checks to timeout. I added `DEB_BUILD_OPTIONS=nocheck` to `debuild` in f3a351c.

File: share/github-backup-utils/ghe-backup-redis [hunk 1]
```diff
@@ -19,12 +19,25 @@ ghe_remote_version_required "$GHE_HOSTNAME"
...
+    for ((n=0; n<240; n++)); do
+      if [ \$(redis-cli LASTSAVE) -eq \$timestamp ]; then
+          sleep 15
+      else
+        break
+      fi
+      if [ \$n -eq 239 ]; then
+        exit 1
+      fi
     done
```
Comment (rubiojr) [inline share/github-backup-utils/ghe-backup-redis]: This for loop style is a bit more compact and preferred.

```bash
for n in $(seq 240); do
  if [ "\$(redis-cli LASTSAVE)" != "\$timestamp" ]; then
    break
  fi
done
[ "\$(redis-cli LASTSAVE) != "\$timestamp" ] # exits 1 if bgsave didn't work
```

File: share/github-backup-utils/ghe-backup-redis [hunk 2]
```diff
@@ -19,12 +19,25 @@ ghe_remote_version_required "$GHE_HOSTNAME"
...
-    redis-cli BGSAVE 1>/dev/null
-    while [ \$(redis-cli LASTSAVE) -eq \$timestamp ]; do
-        sleep 1
+    for ((n=0; n<10; n++)); do
+      if redis-cli BGSAVE | grep -q ERR; then
+        sleep 15
+      else
+        break
+      fi
+    done
```
Comment (rubiojr) [inline share/github-backup-utils/ghe-backup-redis]: This for loop style is preferred:

```bash
for i in $(seq 10); do
  if ! redis-cli BGSAVE | grep -q ERR; then
    break
  fi
  sleep 15
done
```

**Comment by buckelij** [comment]
@rubiojr made the style changes and re-enabled the test. Something must have been going on with CI earlier -- the travis-ci checks were passing but the backup-utils test was timing out during the deb package build.

File: share/github-backup-utils/ghe-backup-redis [hunk 3]
```diff
@@ -19,13 +19,23 @@ ghe_remote_version_required "$GHE_HOSTNAME"
...
-    while [ \$(redis-cli LASTSAVE) -eq \$timestamp ]; do
-        sleep 1
+    for i in \$(seq 10); do
+      if ! redis-cli BGSAVE | grep -q ERR; then
+        break
+      fi
+      sleep 15
     done
+    for n in \$(seq 240); do
+      sleep 1
```
Comment (buckelij) [inline share/github-backup-utils/ghe-backup-redis]: I added a sleep here and took a second off the 2nd sleep. The fake `redis-cli` used in the test environment increments every second, but this was running through too fast and adding an extra 15 seconds to every backup-utils run. That was adding up and causing the tests to time out.

**Comment by buckelij** [comment]
Ready for review again @rubiojr

File: share/github-backup-utils/ghe-backup-redis (hunk 3 above)
Comment (rubiojr) [inline share/github-backup-utils/ghe-backup-redis]: @buckelij any reason why we can't just replace `240` with `3600` to simplify things?

Comment (rubiojr) [inline share/github-backup-utils/ghe-backup-redis]: And leave only the `sleep 1`, that is.

Comment (buckelij) [inline share/github-backup-utils/ghe-backup-redis]: yeah, that's fine. Fixed in 5627d6d.

**Comment by rubiojr** [comment]
:bow:

**Pull request merged**